hypertrace-gradle-publish-maven-central-plugin/README.md

```diff
@@ -0,0 +1,66 @@
+# Hypertrace Publish Maven Central Plugin
+
+###### org.hypertrace.publish-maven-central-plugin
+
+### Purpose
+This plugin configures the target project to publish its java artifacts to maven central. 
+It uses the `maven-publish` and `signing` plugins internally to do this.
+By default, it will publish to https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/. 
+It also adds `javadocJar` and `sourcesJar` tasks to the project.
+The following credentials must be provided as gradle properties in the default configuration:
+- ossrhUsername
+- ossrhPassword
+
+For signing artifacts, following properties must be provided as gradle properties:
+- signingKey
+- signingPassword
+
+Each property described below can be configured in the DSL. The default values are shown for each property,
+all of which, with the exception of license, can be omitted if left unchanged.
+```kotlin
+ hypertracePublishMavenCentral {
+    url // The URL for the publication represented by the POM.
+    scmConnection // The connection URL of the SCM.
+    scmDeveloperConnection // The developer connection URL of the SCM
+    scmUrl // The browsable repository URL of the SCM
+    licenses // Configures the licenses for the publication represented by the POM
+    developers // Configures the developers for the publication represented by the POM
+  }
+```
+
+Currently supported publications:
+- `java-library`: For projects applying `java-library`, the `java` component (i.e. the jar) will be registered as a publication
+### Example
+#### Root project
+
+```kotlin
+// Specify the publish version but don't apply it
+plugins {
+  id("org.hypertrace.publish-maven-central-plugin") version "<version>" apply false
+}
+
+subprojects {
+    pluginManager.withPlugin("org.hypertrace.publish-maven-central-plugin") {
+        configure<org.hypertrace.gradle.publishing.HypertracePublishMavenCentralExtension> {
+            url.set("https://www.hypertrace.org")
+            scmConnection.set("scm:git:git://github.com/hypertrace/hypertrace.git")
```

Comment:
So the nice thing about having our own extension here is we can limit this down to the things we actually want configured, and then push those values into the appropriate spots in the POM.

- url should always be hypertrace, so we should omit it from configuration (if we eventually have that requirement, could always add it - but for now, we want to make this as simple to use as possible)
- All three scm values are derivable - we could just ask for like repo name (since that's the only part that changes between repos)


if we're able to make those changes, using it just becomes:
```kts
subprojects {
    pluginManager.withPlugin("org.hypertrace.publish-maven-central-plugin") {
        configure<org.hypertrace.gradle.publishing.HypertracePublishMavenCentralExtension> {
          repoName.set("some-hypertrace-repo")
        }
    }
}
```

Reply:
Default value of url is https://www.hypertrace.org.

Using this plugin requires two values:
1. license
2. repoName